docs/conandata_yml_format.md

```diff
@@ -0,0 +1,220 @@
+```
+
+#### several source code archives
+
+certain projects may include multiple tarballs as a part of release, [OpenCV](https://opencv.org/) is an example which includes auxillary [contrib](https://github.com/opencv/opencv_contrib) archive:
+```
+sources:
+  "4.5.0":
+    - sha256: dde4bf8d6639a5d3fe34d5515eab4a15669ded609a1d622350c7ff20dace1907
+      url: https://github.com/opencv/opencv/archive/4.5.0.tar.gz
+    - sha256: a65f1f0b98b2c720abbf122c502044d11f427a43212d85d8d2402d7a6339edda
+      url: https://github.com/opencv/opencv_contrib/archive/4.5.0.tar.gz
+```
+
+#### different source code archives per configuration
+
+this is the most advanced and sophisticated use-case, fortunately, rarely encountered in the wild. project may provide different sources for different platforms for awkward reasons, it could be expressed as:
+
+```
+sources:
+  "0066":
+    "Macos":
+      "apple-clang":
+        "x86":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_32bit/packages/cspice.tar.Z"
+            sha256: "9a4b5f674ea76821c43aa9140829da4091de646ef3ce40fd5be1d09d7c37b6b3"
+        "x86_64":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_64bit/packages/cspice.tar.Z"
+            sha256: "f5d48c4b0d558c5d71e8bf6fcdf135b0943210c1ff91f8191dfc447419a6b12e"
+```
+
+this approach requires a special code within [source](https://docs.conan.io/en/latest/reference/conanfile/methods.html#source) method to handle.
+
+## patches
+
+sometimes sources provided by project require patching for various reasons. `conandata.yml` may include information about patches.
+
+### patch_file
+
+patch file might be commited to the conan-center-index, near to the conanfile (usuaully, into the `patches` sub-directory). such patch files usually have either `.diff` or `.patch` extension.
+the recommented way to generate such patches is [git format-patch](https://git-scm.com/docs/git-format-patch). the path to the patch is relative to the directory containing `conandata.yml` and `conanfile.py`.
+
+### base_path
+
+specifies a sub-directory in project's sources to apply patch. directory is relative to the [source_folder](https://docs.conan.io/en/latest/reference/conanfile/attributes.html?highlight=source_folder#source-folder). usually, it would be a `source_subfolder`, but could be a lower-level sub-directory (e.g. if it's a patch for the submodule),
+
+### url
+
+the patch could be taken from the existing resource (e.g. some patches are distributed alongside the source tarball). see `url` section of the `sources`.
+
+### sha256
+
+for the patch downloaded from the Internet, provides a hashsum. see `sha256` section of the `sources`.
+
+### sha1
+
+for the patch downloaded from the Internet, provides a hashsum. see `sha1` section of the `sources`.
+
+### md5
+
+for the patch downloaded from the Internet, provides a hashsum. see `md5` section of the `sources`.
+
+### patch_type
```

Comment:
Here I would just enumerate the type of patches with a very short description and link to `docs/policy_patching.md` file where everything should be explained. We will update that document mentioning the explicit name (official, vulnerability, backport,...) for each patch type.

Reply:
I had an exactly the opposite idea - referencing this document from the `policy_patching`